specification/sbi_atee.adoc

```diff
@@ -13,6 +43,858 @@ running a VMM).
+
+-----
+Reclaims `num_pages` of confidential memory starting at `base_page_address`.
+The pages must not be currently assigned to an active TVM.
+
+The `base_page_address` must be page-aligned. The permissible values for the `page_type`
+parameter appear below.
+[source, C]
+-------------------
+enum tsm_page_type {
+    /* 4KiB */
+    PAGE_4K = 0,
+    /* 2 MiB */
+    PAGE_2MB = 1,
+    /* 1 GiB */
+    PAGE_1GB = 2,
+    /* 512 GiB */
+    PAGE_512GB = 3,
+}
+-------------------
+
+The possible error codes returned in `sbiret.error` are shown below.
+
+[#table_tee_tsm_reclaim_pages_errors]
+.TEE TSM Reclaim Pages
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code            | Description
+| SBI_SUCCESS           | The operation completed successfully
+| SBI_ERR_INVALID_PARAM | One or more parameters were invalid.
+|===
+
+=== Function: TEE TSM Initiate Fence (FID #14)
+[source, C]
+-----
+struct sbiret sbi_tee_tsm_initiate_fence(void);
+-----
+Initiates a TLB invalidation sequence for all pages marked for conversion via
+calls to `sbi_tee_tsm_convert_pages` between the previous `sbi_tee_tsm_initiate_fence()`
+and now. The TLB invalidation sequence is completed when `sbi_tee_tsm_local_fence()` has
+been invoked on all other CPUs. An error is returned if a TLB invalidation sequence is
+already in progress.
+
+The possible error codes returned in `sbiret.error` are shown below.
+
+[#table_sbi_tee_tsm_initiate_fence_errors]
+.TEE TSM Initiate Fence
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code            | Description
+| SBI_SUCCESS           | The operation completed successfully
+| SBI_ERR_INVALID_PARAM | A fence operation is already in progress
+|===
+
+=== Function: TEE TSM Local Fence (FID #15)
+[source, C]
+-----
+struct sbiret sbi_tee_tsm_local_fence(void);
+-----
+Invalidates TLB entries for all pages pending conversion by an in-progress TLB
+invalidation operation on the local CPU.
+
+The possible error codes returned in `sbiret.error` are shown below.
+
+[#table_sbi_tee_tsm_local_fence_errors]
+.TEE TSM Local Fence
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code            | Description
+| SBI_SUCCESS           | The operation completed successfully
+|===
+
+=== Function: TEE TVM Get VCPU Register (FID #16)
+[source, C]
+-----
+struct sbiret sbi_tee_tvm_get_vcpu_register(unsigned long tvm_guest_id,
+                                            unsigned long tvm_vcpu_id,
+                                            unsigned long vcpu_register);
+-----
+Gets the value for the register identified by register in the VCPU identified
+by `tvm_vcpu_id` for the TVM specified by `tvm_guest_id`. The `tvm_guest_id is
+the unique identifier returned by a previous call to`sbi_tee_create_tvm()`.
+
+The permissible values for `tvm_vcpu_register`, and the associated semantics are listed below.
+[source, C]
+-----
+enum tvm_vcpu_register {
```

Comment:
It doesn't have to be defined twice. It would also make sense to keep sbi_tee_tvm_get_vcpu_register/sbi_tee_tvm_set_vcpu_register together.